packages/debug/src/browser/debug-session.tsx

```diff
@@ -308,59 +312,44 @@ export class DebugSession implements CompositeTreeElement {
         await this.updateThreads(undefined);
     }
 
-    protected terminated = false;
-    async terminate(restart?: boolean): Promise<void> {
-        if (!this.terminated && this.capabilities.supportsTerminateRequest && this.configuration.request === 'launch') {
-            this.terminated = true;
-            await this.connection.sendRequest('terminate', { restart });
-            if (!await this.exited(1000)) {
-                await this.disconnect(restart);
-            }
-        } else {
-            await this.disconnect(restart);
-        }
+    canTerminate(): boolean {
```

Comment:
Why would you want it to be a property?

Reply:
No strong reason, just neater as `if (this.canTerminate) ...`

Reply:
Personally, I'm not a big fan of casual use of property getters and setters for two reasons:

1. It doesn't play well with JSON serialisation
2. It may hide arbitrary complexity behind something that looks simple like `foo.bar = true`. 

For those reasons writing property getters and setters need justification, IMO, for example to ensure internal consistency of an object or implementing some "magic" that can't be achieved otherwise. Not having to type `()` is not sufficient reason, IMO, but then, I don't share the industry's obsession with typing less ;-)